Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp

```diff
@@ -1171,22 +1171,29 @@ RefPtr<WebCore::ScrollingCoordinator> WebChromeClient::createScrollingCoordinato
 #endif
 
 #if PLATFORM(MAC)
-std::unique_ptr<ScrollbarsController> WebChromeClient::createScrollbarsController(Page& corePage, ScrollableArea& area) const
+void WebChromeClient::ensureScrollbarsController(Page& corePage, ScrollableArea& area) const
 {
     auto page = protectedPage();
     ASSERT(page->corePage() == &corePage);
-    
-    if (area.mockScrollbarsControllerEnabled())
-        return nullptr;
+    auto* currentScrollbarsController = area.rawScrollbarsController();
+    if (area.mockScrollbarsControllerEnabled() || (currentScrollbarsController && is<RemoteScrollbarsController>(currentScrollbarsController)))
```

Comment:
It's only correct to keep the existing `RemoteScrollbarsController` if `area.usesAsyncScrolling()` is still true, right? If a ScrollableArea goes from async to non-async, we have to change the ScrollbarsController back.

Reply:
Fixed.